bc_obps/registration/tests/test_01_models.py

```diff
@@ -505,7 +512,110 @@ def test_field_labels_and_max_lengths_and_multiple_relations(self):
+                )
+
+    def test_generate_unique_tax_exemption_id_existing_id(self):
+        # Case: Operation already has a tax exemption ID
+        existing_id = "22-0001"  # Example existing ID for the current year
+        self.test_operation.tax_exemption_id = existing_id
+        generated_id = self.test_operation.generate_unique_tax_exemption_id()
+        self.assertEqual(generated_id, existing_id, "Should return the existing ID.")
+
+    def test_generate_unique_tax_exemption_id_no_existing_id(self):
+        # Case: No existing tax exemption ID
+        self.test_operation.tax_exemption_id = None
+        self.test_operation.generate_unique_tax_exemption_id()
+        current_year = datetime.now().year % 100
+        expected_id = f"{current_year:02d}-0001"  # Assuming the first ID for the year
+        self.assertEqual(
+            self.test_operation.tax_exemption_id, expected_id, "Should generate a new ID for the current year."
+        )
+
+    def test_generate_unique_tax_exemption_id_multiple_existing_ids_same_year(self):
+        # Case: Multiple existing tax exemption IDs for the current year
+        existing_ids = ["23-0002", "23-0003", "23-0001"]
+        Operation.objects.bulk_create(
+            [
+                Operation(
+                    name="test",
+                    type="test",
+                    naics_code=NaicsCode.objects.first(),
+                    operator=Operator.objects.first(),
+                    tax_exemption_id=existing_id,
+                )
+                for existing_id in existing_ids
+            ]
+        )
+
+        self.test_operation.tax_exemption_id = None
+        self.test_operation.generate_unique_tax_exemption_id()
+        current_year = datetime.now().year % 100
+        expected_id = f"{current_year:02d}-0004"
+        self.assertEqual(
+            self.test_operation.tax_exemption_id, expected_id, "Should generate a new ID for the current year."
+        )
+
+    def test_generate_unique_tax_exemption_id_multiple_existing_ids_different_year(self):
+        # Case: Multiple existing tax exemption IDs for the current year
+        existing_ids = ["22-0001", "22-0002", "22-0003"]
+        Operation.objects.bulk_create(
+            [
+                Operation(
+                    name="test",
+                    type="test",
+                    naics_code=NaicsCode.objects.first(),
+                    operator=Operator.objects.first(),
+                    tax_exemption_id=existing_id,
+                )
+                for existing_id in existing_ids
+            ]
+        )
+
+        self.test_operation.tax_exemption_id = None
+        self.test_operation.generate_unique_tax_exemption_id()
+        current_year = datetime.now().year % 100
+        expected_id = f"{current_year:02d}-0001"
+        self.assertEqual(
+            self.test_operation.tax_exemption_id, expected_id, "Should generate a new ID for the current year."
+        )
```

Comment:
LOVE how thorough you've been with these test cases! Thanks Sepehr, you're helping me sleep at night!